scenario/mocking.py

```diff
@@ -73,26 +73,40 @@ class ActionMissingFromContextError(Exception):
 
 
 class _MockExecProcess:
-    def __init__(self, command: Tuple[str, ...], change_id: int, out: "ExecOutput"):
+    def __init__(
+        self,
+        command: Tuple[str, ...],
+        change_id: int,
+        exec: "Exec",
+    ):
         self._command = command
         self._change_id = change_id
-        self._out = out
+        self._exec = exec
         self._waited = False
-        self.stdout = StringIO(self._out.stdout)
-        self.stderr = StringIO(self._out.stderr)
+        self.stdout = StringIO(self._exec.stdout)
+        self.stderr = StringIO(self._exec.stderr)
+        # You can't pass *in* the stdin, the charm is responsible for that.
```

Comment:
`if exec.stdin: raise Something`?

Reply:
Yes, it's probably best to do that. Or to have the consistency checker fail it? The straight exception is probably the better choice, I think.